File: tests/nodejs/cases/case_115_thread_handoff_late_02.js
/**
 * Task case 115: thread_handoff_late_02 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case115ThreadHandoffLate02(input) {
  const taskName = 'thread_handoff_late_02';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: async microtask captures payload beyond return edge.
  queueMicrotask(() => retainedAudit.push(payload));
  return 'ok';
}

module.exports = {
  case115ThreadHandoffLate02,
};

File: tests/go/cases/case_115_thread_handoff_late_02.go
package escape_tests

var retainedCase115 = []map[string]string{}

func Case115ThreadHandoffLate02(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "thread_handoff_late_02",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "thread_handoff_late_02:" + raw,
	}
	// ESCAPE: goroutine captures payload and appends to retained sink.
	go func(p map[string]string) {
		retainedCase115 = append(retainedCase115, p)
	}(payload)
	return "ok"
}
